patches/api/0469-Add-CartographyItemEvent.patch

```diff
@@ -0,0 +1,42 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Janet Blackquill <uhhadd@gmail.com>
+Date: Sun, 7 Apr 2024 16:51:04 -0400
+Subject: [PATCH] Add CartographyItemEvent
+
+Similar to SmithItemEvent, but for cartography tables.
+
+diff --git a/src/main/java/io/papermc/paper/event/player/CartographyItemEvent.java b/src/main/java/io/papermc/paper/event/player/CartographyItemEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..6dd6910c4f3537b2a9f6140b042b3679ccb4d864
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/CartographyItemEvent.java
+@@ -0,0 +1,29 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.inventory.InventoryView;
++import org.bukkit.inventory.CartographyInventory;
++import org.bukkit.event.inventory.ClickType;
++import org.bukkit.event.inventory.InventoryType;
++import org.bukkit.event.inventory.InventoryAction;
++import org.bukkit.event.inventory.InventoryClickEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when the recipe of an Item is completed inside a cartography table.
++ */
++public class CartographyItemEvent extends InventoryClickEvent {
++
++    public CartographyItemEvent(@NotNull InventoryView view, @NotNull InventoryType.SlotType type, int slot, @NotNull ClickType click, @NotNull InventoryAction action) {
++        super(view, type, slot, click, action);
++    }
++
++    public CartographyItemEvent(@NotNull InventoryView view, @NotNull InventoryType.SlotType type, int slot, @NotNull ClickType click, @NotNull InventoryAction action, int key) {
```

Comment:
Add `@ApiStatus.Internal` to both these constructors. Also, I don't know if other events of this type have them, but we can add Preconditions I think to the constructors to ensure that the Inventory passed in is a CartographyInventory instead of a potential error when someone uses the getter and it throws a ClassCastException